temporalcli/commandsgen/docs.go

```diff
@@ -75,27 +76,38 @@ func (w *docWriter) writeCommand(c *Command) {
 }
 
 func (w *docWriter) writeSubcommand(c *Command) {
-	fileName := strings.Split(c.FullName, " ")[1]
-	subCommand := strings.Join(strings.Split(c.FullName, " ")[2:], "")
-	w.fileMap[fileName].WriteString("## " + subCommand + "\n\n")
+	fileName := c.FileName
+	prefix := strings.Repeat("#", c.Depth)
+	w.fileMap[fileName].WriteString(prefix + " " + c.LeafName + "\n\n")
 	w.fileMap[fileName].WriteString(c.Description + "\n\n")
-	w.fileMap[fileName].WriteString("Use the following options to change the behavior of this command.\n\n")
-
-	// gather options from command and all options aviailable from parent commands
-	var allOptions = make([]Option, 0)
-	for _, options := range w.optionsStack {
-		allOptions = append(allOptions, options...)
-	}
 
-	// alphabetize options
-	sort.Slice(allOptions, func(i, j int) bool {
-		return allOptions[i].Name < allOptions[j].Name
-	})
+	if isLeafCommand(c) {
+		w.fileMap[fileName].WriteString("Use the following options to change the behavior of this command.\n\n")
 
-	for _, option := range allOptions {
-		w.fileMap[fileName].WriteString(fmt.Sprintf("## %s\n\n", option.Name))
-		w.fileMap[fileName].WriteString(option.Description + "\n\n")
+		// gather options from command and all options aviailable from parent commands
+		var allOptions = make([]Option, 0)
+		for _, options := range w.optionsStack {
+			allOptions = append(allOptions, options...)
+		}
 
+		// alphabetize options
+		sort.Slice(allOptions, func(i, j int) bool {
+			return allOptions[i].Name < allOptions[j].Name
+		})
+
+		for _, option := range allOptions {
+			w.fileMap[c.FileName].WriteString(fmt.Sprintf("**--%s**\n\n", option.Name))
```

Comment:
For flags that take an argument (as opposed to just being a boolean), we should probably show that here like we do in the `--help` output:

```
  -n, --namespace string    Temporal Service Namespace. (default "default")
```

So printing something like `**--namespace** _string_` would be nice.

Reply:
Agree, that would be a nice addition.